server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -1780,11 +1795,16 @@ public void deleteSegments(final Set<DataSegment> segments)
           public Void inTransaction(Handle handle, TransactionStatus transactionStatus)
           {
             int segmentSize = segments.size();
-            String dataSource = "";
+            String dataSource = segments.stream().findAny().map(DataSegment::getDataSource).orElse("?");
```

Comment:
Do we need this `.orElse("?")`? Also,  I think we can move this line and the query declaration outside the scope of the transaction.

Reply:
Yeah, better to just check `segments.isEmpty()` at the start of the `deleteSegments` and return if it is empty. Then, we wouldn't need the `orElse()`.

Reply:
Sure; removed. I had retained this for compatibility with existing code.